ste/sender-blobSymlinks.go

```diff
@@ -0,0 +1,131 @@
+	if blockBlobTierOverride != common.EBlockBlobTier.None() {
+		destBlobTier = blockBlobTierOverride.ToAccessTierType()
+	}
+
+	var out sender
+	ssend := blobSymlinkSender{
+		jptm:             jptm,
+		sip:              sip,
+		destBlockBlobURL: destBlockBlobURL,
+		metadataToApply:  props.SrcMetadata.Clone().ToAzBlobMetadata(), // We're going to modify it, so we should clone it.
+		headersToApply:   props.SrcHTTPHeaders.ToAzBlobHTTPHeaders(),
+		blobTagsToApply:  props.SrcBlobTags.ToAzBlobTagsMap(),
+		cpkToApply:       common.ToClientProvidedKeyOptions(jptm.CpkInfo(), jptm.CpkScopeInfo()),
+		destBlobTier:     destBlobTier,
+	}
+	fromTo := jptm.FromTo()
+	if fromTo.IsUpload() {
+		out = &dummySymlinkUploader{ssend}
+	} else {
+		out = &dummySymlinkS2SCopier{ssend}
+	}
+
+	return out, nil
+}
+
+func (s *blobSymlinkSender) SendSymlink(linkData string) error {
+	err := s.getExtraProperties()
+	if err != nil {
+		return fmt.Errorf("when getting additional folder properties: %w", err)
+	}
+	s.metadataToApply["is_symlink"] = "true" // For BlobFuse
+
+	_, err = s.destBlockBlobURL.Upload(s.jptm.Context(), strings.NewReader(linkData), s.headersToApply, s.metadataToApply, azblob.BlobAccessConditions{}, s.destBlobTier, s.blobTagsToApply, s.cpkToApply, azblob.ImmutabilityPolicyOptions{})
+	return err
+}
+
+// ===== Implement sender so that it can be returned in newBlobUploader. =====
+/*
+	It's OK to just panic all of these out, as they will never get called in a symlink transfer.
+*/
+
+func (s *blobSymlinkSender) ChunkSize() int64 {
+	panic("this sender only sends symlinks.")
+}
+
+func (s *blobSymlinkSender) NumChunks() uint32 {
+	panic("this sender only sends symlinks.")
+}
+
+func (s *blobSymlinkSender) RemoteFileExists() (bool, time.Time, error) {
+	panic("this sender only sends symlinks.")
+}
+
+func (s *blobSymlinkSender) Prologue(state common.PrologueState) (destinationModified bool) {
+	panic("this sender only sends symlinks.")
+}
+
+func (s *blobSymlinkSender) Epilogue() {
+	panic("this sender only sends symlinks.")
+}
+
+func (s *blobSymlinkSender) Cleanup() {
+	panic("this sender only sends symlinks.")
+}
+
+func (s *blobSymlinkSender) GetDestinationLength() (int64, error) {
+	panic("this sender only sends symlinks.")
+}
+
+// implement uploader to handle commonSenderCompletion
+
+type dummySymlinkUploader struct {
+	blobSymlinkSender
+}
+
+func (d dummySymlinkUploader) GenerateUploadFunc(chunkID common.ChunkID, blockIndex int32, reader common.SingleChunkReader, chunkIsWholeFile bool) chunkFunc {
+	panic("this sender only sends folders.")
+}
+
+func (d dummySymlinkUploader) Md5Channel() chan<- []byte {
+	panic("this sender only sends folders.")
+}
+
+// ditto for s2sCopier
+
+type dummySymlinkS2SCopier struct {
+	blobSymlinkSender
+}
+
+func (d dummySymlinkS2SCopier) GenerateCopyFunc(chunkID common.ChunkID, blockIndex int32, adjustedChunkSize int64, chunkIsWholeFile bool) chunkFunc {
+	// TODO implement me
+	panic("implement me")
```

Comment:
Missing implementation?

Reply:
Much like the above dummy sender, most of the functions are panicked out as there is no need nor use for them. They are solely implemented for the sake of copying symlinks.